test/inherit-parameters.js

```diff
@@ -0,0 +1,234 @@
+  context('Query Parameter', () => {
+    it('on Operation', (done) => {
+      const source = makeSource('/');
+      source.paths['/'].get.parameters.push(makeParameter('test', 'query'));
+
+      doParse(source, done, (result) => {
+        expect(result.resource.href).to.be.equal('/');
+        expect(result.transition.href).to.be.equal('/{?test}');
+      });
+    });
+
+    it('on Path', (done) => {
+      const source = makeSource('/');
+      source.paths['/'].parameters.push(makeParameter('test', 'query'));
+
+      doParse(source, done, (result) => {
+        expect(result.resource.href).to.be.equal('/{?test}');
+        expect(result.transition.href).to.not.exist;
+      });
+    });
+
+    it('on Path and Operation', (done) => {
+      const source = makeSource('/');
+      source.paths['/'].parameters.push(makeParameter('test', 'query'));
+      source.paths['/'].get.parameters.push(makeParameter('foo', 'query'));
+
+      doParse(source, done, (result) => {
+        expect(result.resource.href).to.be.equal('/{?test}');
+        expect(result.transition.href).to.be.equal('/{?test,foo}');
+      });
+    });
+  });
+
+  context('Path Parameter', () => {
+    it('on Path', (done) => {
+      const source = makeSource('/{test}');
+      source.paths['/{test}'].parameters.push(makeParameter('test', 'path'));
+
+      doParse(source, done, (result) => {
+        expect(result.resource.href).to.be.equal('/{test}');
+        expect(result.resource.hrefVariables.keys()).to.deep.equal(['test']);
+
+        expect(result.transition.href).to.not.exist;
+        expect(result.transition.hrefVariables).to.not.exist;
+      });
+    });
+
+    it('on Operation', (done) => {
+      const source = makeSource('/{test}');
+      source.paths['/{test}'].get.parameters.push(makeParameter('test', 'path'));
+
+      doParse(source, done, (result) => {
+        expect(result.resource.href).to.be.equal('/{test}');
+        expect(result.resource.hrefVariables).to.not.exist;
+
+        expect(result.transition.href).to.not.exist;
+        expect(result.transition.hrefVariables.keys()).to.deep.equal(['test']);
+      });
+    });
+
+    it('on Path and Operation', (done) => {
+      const source = makeSource('/{test}');
+      source.paths['/{test}'].parameters.push(makeParameter('test', 'path'));
+      source.paths['/{test}'].get.parameters.push(makeParameter('test', 'path'));
+
+      doParse(source, done, (result) => {
+        expect(result.resource.href).to.be.equal('/{test}');
+        expect(result.transition.href).to.not.exist;
+
+        expect(result.resource.hrefVariables.keys()).to.deep.equal(['test']);
+        expect(result.transition.hrefVariables.keys()).to.deep.equal(['test']);
+      });
+    });
+  });
+
+  context('Header Parameter', () => {
+    it('on Path', (done) => {
+      const source = makeSource('/');
+      source.paths['/'].parameters.push(makeParameter('test', 'header'));
+
+      doParse(source, done, (result) => {
```

Comment:
@pksunkara as we spoke earlier, for this case is not created sourceMap, please, look at it